### PR TITLE
[RW-11656][risk=no] Added missing app contract expectations

### PR DESCRIPTION
**Pull request by evrii**
Added missing contract expectations.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: api/src/contracttest/java/org/pmiops/workbench/consumer/AppsApiTest.java [hunk 1]
```diff
@@ -78,14 +85,26 @@ static LeonardoCreateAppRequest createAppRequest() {
...
     request.setDescriptorPath("descriptor/path");
     request.setDiskConfig(
-        new LeonardoPersistentDiskRequest().diskType(LeonardoDiskType.SSD).size(100));
-    request.setCustomEnvironmentVariables(new ArrayList<>());
+        new LeonardoPersistentDiskRequest()
+            .diskType(LeonardoDiskType.SSD)
+            .size(100)
+            .name("mockDisk"));
+    request.setCustomEnvironmentVariables(Map.ofEntries(entry("key1", "value1")));
     request.setExtraArgs(new ArrayList<>());
```
Comment (freemabd) [inline api/src/contracttest/java/org/pmiops/workbench/consumer/AppsApiTest.java:93]: nit: This seems unnecessary and confusing. Do you have to set this upon initialization to an empty list?

File: api/src/contracttest/java/org/pmiops/workbench/consumer/AppsApiTest.java [hunk 2]
```diff
@@ -290,7 +285,12 @@ RequestResponsePact listAppsByProject(PactDslWithProvider builder) {
         .willRespondWith()
         .status(200)
         .headers(contentTypeJsonHeader)
-        .body(newJsonArray(array -> {}).build())
+        .body(
+            newJsonArray(
+                    apps -> {
+                      apps.object(this::applyAppExpectations);
+                    })
+                .build())
```
Comment (freemabd) [inline api/src/contracttest/java/org/pmiops/workbench/consumer/AppsApiTest.java:293]: nit: Don't need the curly braces 
```suggestion
        .body(newJsonArray(apps -> apps.object(this::applyAppExpectations)).build())
```

File: api/src/contracttest/java/org/pmiops/workbench/consumer/AppsApiTest.java [hunk 3]
```diff
@@ -46,16 +48,21 @@ class AppsApiTest {
...
-                  diskConfig.stringType("diskType", LeonardoDiskType.SSD.name());
+                  diskConfig.stringType("diskType", LeonardoDiskType.SSD.toString());
                   diskConfig.numberType("size");
+                  diskConfig.stringType("name", "mock-disk");
+                });
+            body.object(
+                "customEnvironmentVariables",
+                customEnvironmentVariables -> {
+                  customEnvironmentVariables.stringType("key1");
                 });
```
Comment (freemabd) [inline api/src/contracttest/java/org/pmiops/workbench/consumer/AppsApiTest.java:65]: nit: Don't need the curly braces here: 
```suggestion
                customEnvironmentVariables -> customEnvironmentVariables.stringType("key1"));
```

**Review by freemabd — APPROVED** [review]
Looks good. I left a few comments

**Comment by evrii** [comment]
Thanks for the feedback @freemabd!

**Pull request merged**